### ISSUE TITLE
Test failure building for NixOS

### ISSUE THREAD

**Issue by greg-hellings** [xonsh/xonsh#5569]
## Current Behavior

Trying to build the released 0.17.0 version for nixpkgs, I have gotten tests running with the exception of test_spec_modifier_alias_output_format. It appears to be searching for the `echo` command, which seems like it's typically a built-in function. For some reason, during this particular test, it is searching instead for echo to be a system executable.

<details>

=================================== FAILURES ===================================
____________________ test_spec_modifier_alias_output_format ____________________

xession = <xonsh.built_ins.XonshSession object at 0x7fffe842cc10>

    @skip_if_on_windows
    def test_spec_modifier_alias_output_format(xession):
        class SpecModifierOutputLinesAlias(SpecModifierAlias):
            def on_modifer_added(self, spec):
                spec.output_format = "list_lines"
    
        xession.aliases["xlines"] = SpecModifierOutputLinesAlias()
    
        cmds = [["xlines", "echo", "1\n2\n3"]]
        specs = cmds_to_specs(cmds, captured="stdout")
        (p := _run_command_pipeline(specs, cmds)).end()
>       assert p.output == ["1", "2", "3"]
E       AssertionError: assert [] == ['1', '2', '3']
E         
E         Right contains 3 more items, first extra item: '1'
E         Use -v to get more diff

tests/procs/test_specs.py:331: AssertionError
----------------------------- Captured stderr call -----------------------------
xonsh: For full traceback set: $XONSH_SHOW_TRACEBACK = True
xonsh: subprocess mode: command not found: 'echo'

</details>

## Expected Behavior

Tests build and succeed in nixpkgs/NixOS builds if the appropriate set of dependency packages is present.

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by SamLukeYes** [comment]
The test was renamed to `test_spec_decorator_alias_output_format` in https://github.com/xonsh/xonsh/pull/5579, and is still failing when building xonsh 0.18.2 in https://github.com/NixOS/nixpkgs/pull/325813

**Comment by SamLukeYes** [comment]
> // Just thought: what about adding Nix to [xonsh CI](https://github.com/xonsh/xonsh/blob/938db85a8733eb15be76d8b599c0d6dfc68b7cac/.github/workflows/test.yml#L33)? Is it possible?

Though I'm a noob at GitHub actions setup, I know there are some actions that make use of Nix, e.g. [cachix/install-nix-action](https://github.com/cachix/install-nix-action). Also consider [Garnix CI](https://garnix.io/), which builds packages and checks in a Flake hosted on GitHub. Additionally, adding `flake.nix` to this repo would allow Nix Flakes users to use or test the latest main branch of Xonsh easily, though its packaging might need to be simplified, compared to the one in Nixpkgs, which is still a headache to me :rofl:

**Comment by greg-hellings** [comment]
Yes, we can help add Nix tests and builds directly to this repository. I'm on a trip right now, but I would be happy to help with this over the next few days.

**Comment by SamLukeYes** [comment]
<img width="3072" height="1664" alt="Image" src="https://github.com/user-attachments/assets/8b44cab4-ecab-468d-b1e6-e07d66291798" />

It seems related xonsh tests use hard-coded (default?) environment variable `PATH`, and unfortunately `echo` doesn't exist in the hard-coded `PATH` on NixOS or inside nix build sandbox.

**Comment by anki-code** [comment]
@SamLukeYes it's here:

https://github.com/xonsh/xonsh/blob/b65a981896ac4c7956700a3dfdd68ba5d5f1a68c/xonsh/platform.py#L517-L541

What is the right way to form this on Nix? PR is welcome!

**Comment by SamLukeYes** [comment]
Just made a simple package with nix to see how `PATH` looks like inside nix build sandbox:

```xsh
@ cat test.nix
with import <nixpkgs> {};

runCommand "echo-path-test" {
  buildInputs = [ which ];
} ''
  mkdir -p $out
  echo $PATH > $out/PATH
  which echo > $out/echo-path
''

@ nix-build test.nix
this derivation will be built:
  /nix/store/63pff2yiv9fi9m5lssijhzjz66d5xgw2-echo-path-test.drv
building '/nix/store/63pff2yiv9fi9m5lssijhzjz66d5xgw2-echo-path-test.drv'...
/nix/store/09bb6lbkr4mqr7n96ggb1agfgi5j379c-echo-path-test

@ cat result/PATH
/nix/store/590yx3aynyhs48jyk8ip37fk1mjqfhkb-patchelf-0.15.2/bin:/nix/store/r4k27sfd5psf3if9imbsjfcgba3g6lx5-which-2.23/bin:/nix/store/hlxw2q9qansq7bn52xvlb5badw3z1v8s-coreutils-9.10/bin:/nix/store/b3rx5wac9hhfxn9120xkcvdwj51mc9z2-findutils-4.10.0/bin:/nix/store/icrrz26xbyp293kagrlkab1bhc6gra0r-diffutils-3.12/bin:/nix/store/wv7qq5yb8plyhxji9x3r5gpkyfm2kf29-gnused-4.9/bin:/nix/store/8laf6k81j9ckylrigj3xsk76j69knhvl-gnugrep-3.12/bin:/nix/store/gf7b4yz4vhd0y2hnnrimhh875ghwzzzj-gawk-5.3.2/bin:/nix/store/isva9q9zx3frx6hh6cnpihh1kd2bx6bk-gnutar-1.35/bin:/nix/store/w1n7yp2vnldr395hbwbcaw9sflh413bm-gzip-1.14/bin:/nix/store/x8l7qzpab2gpdrp89g48mxlrsiz4f0gm-bzip2-1.0.8-bin/bin:/nix/store/0xw6y53ijaqwfd9c99wyaqiinychzv1f-gnumake-4.4.1/bin:/nix/store/2hjsch59amjs3nbgh7ahcfzm2bfwl8zi-bash-5.3p9/bin:/nix/store/8y5jm97n4lyw80gh71yihghbhqc11fdz-patch-2.8/bin:/nix/store/27fx8p4k6098wan3zahdbyj79ndcn03z-xz-5.8.2-bin/bin:/nix/store/p3j7lphwlci13f9w2v4rav6rbvpi80li-file-5.45/bin

@ cat result/echo-path
/nix/store/hlxw2q9qansq7bn52xvlb5badw3z1v8s-coreutils-9.10/bin/echo
```

The path of `echo` depends on which version of `coreutils` is used.

> What is the right way to form this on Nix?

The point is not to use a hard-coded default value of `PATH`. Either use the `PATH` set in the environment, or allow additional paths to be set through something like pytest flags.

**Comment by anki-code** [comment]
@SamLukeYes I've introduced the fix in #6131, please test.
(it will be cool to add Nix to [xonsh test matrix](https://github.com/xonsh/xonsh/blob/b65a981896ac4c7956700a3dfdd68ba5d5f1a68c/.github/workflows/test.yml#L33) some day)
